From a52bda3dba262ef2b3a339341a6f52e67c717030 Mon Sep 17 00:00:00 2001
From: "jaekyu.lee" <jaekyu.lee@lge.com>
Date: Thu, 23 Jan 2025 14:27:52 +0900
Subject: [PATCH] Fix build with liburcu-0.15.0

:Release Notes:
Fix build error

:Detailed Notes:
Fixes:
http://gecko.lge.com:8000/Errors/Details/1095316

../../../git/pmtrace/plugin/pmtraceplugin.cpp:30:91: warning: cannot declare member function 'static QObject* PmTracePlugin::pmTraceProvider(QQmlEngine*, QJSEngine*)' to have static linkage [-fpermissive]
   30 | static QObject *PmTracePlugin::pmTraceProvider(QQmlEngine *engine, QJSEngine *scriptEngine)
      |                                                                                           ^
In file included from TOPDIR/BUILD/work/raspberrypi4_64-webos-linux/qml-webos-components/1.0.0-59/recipe-sysroot/usr/include/lttng/tracepoint-rcu.h:10,
                 from TOPDIR/BUILD/work/raspberrypi4_64-webos-linux/qml-webos-components/1.0.0-59/recipe-sysroot/usr/include/lttng/tracepoint.h:13,
                 from ../../../git/pmtrace/plugin/pmtrace_qmlplugin_provider.h:31,
                 from ../../../git/pmtrace/plugin/QmlPluginTracer.h:22,
                 from ../../../git/pmtrace/plugin/pmtrace.cpp:24:
TOPDIR/BUILD/work/raspberrypi4_64-webos-linux/qml-webos-components/1.0.0-59/recipe-sysroot/usr/include/urcu/compiler.h:205:1: error: template with C linkage
  205 | template <typename T>

:Testing Performed:
Only build tested.

:QA Notes:
No change to image.

:Issues Addressed:
[WRR-17603] Fix build with liburcu-0.15.0

Change-Id: I7467077afb1f7ab27bd75bd70426071c5b82fe32
---
Upstream-Status: Submitted [http://gpro.lge.com/c/webosose/qml-webos-components/+/466486]

 .../plugin/pmtrace_performancelog_provider.h        | 13 ++++++-------
 pmtrace/plugin/pmtrace_qmlplugin_provider.h         | 13 ++++++-------
 2 files changed, 12 insertions(+), 14 deletions(-)

diff --git a/performancelog/plugin/pmtrace_performancelog_provider.h b/performancelog/plugin/pmtrace_performancelog_provider.h
index 1ab811c..73d4348 100644
--- a/performancelog/plugin/pmtrace_performancelog_provider.h
+++ b/performancelog/plugin/pmtrace_performancelog_provider.h
@@ -20,16 +20,15 @@
 #undef TRACEPOINT_INCLUDE
 #define TRACEPOINT_INCLUDE "./pmtrace_performancelog_provider.h"
 
-#ifdef __cplusplus
-extern "C"{
-#endif /*__cplusplus */
-
-
 #if !defined(_PMTRACE_PERFORMANCELOG_PROVIDER_H) || defined(TRACEPOINT_HEADER_MULTI_READ)
 #define _PMTRACE_PERFORMANCELOG_PROVIDER_H
 
 #include <lttng/tracepoint.h>
 
+#ifdef __cplusplus
+extern "C"{
+#endif /*__cplusplus */
+
 /* "message" tracepoint should be used for single event trace points */
 TRACEPOINT_EVENT(
     pmtrace_performancelog,
@@ -49,8 +48,8 @@ TRACEPOINT_EVENT(
     TP_ARGS(char*, text),
     TP_FIELDS(ctf_string(scope, text)))
 
-#endif /* _PMTRACE_PERFORMANCELOG_PROVIDER_H */
-
 #ifdef __cplusplus
 }
 #endif /*__cplusplus */
+
+#endif /* _PMTRACE_PERFORMANCELOG_PROVIDER_H */
diff --git a/pmtrace/plugin/pmtrace_qmlplugin_provider.h b/pmtrace/plugin/pmtrace_qmlplugin_provider.h
index 19ca59a..ceaa32e 100644
--- a/pmtrace/plugin/pmtrace_qmlplugin_provider.h
+++ b/pmtrace/plugin/pmtrace_qmlplugin_provider.h
@@ -20,16 +20,15 @@
 #undef TRACEPOINT_INCLUDE
 #define TRACEPOINT_INCLUDE "./pmtrace_qmlplugin_provider.h"
 
-#ifdef __cplusplus
-extern "C"{
-#endif /*__cplusplus */
-
-
 #if !defined(_PMTRACE_QMLPLUGIN_PROVIDER_H) || defined(TRACEPOINT_HEADER_MULTI_READ)
 #define _PMTRACE_QMLPLUGIN_PROVIDER_H
 
 #include <lttng/tracepoint.h>
 
+#ifdef __cplusplus
+extern "C"{
+#endif /*__cplusplus */
+
 /* "message" tracepoint should be used for single event trace points */
 TRACEPOINT_EVENT(
     pmtrace_qmlplugin,
@@ -63,8 +62,8 @@ TRACEPOINT_EVENT(
     TP_ARGS(char*, text),
     TP_FIELDS(ctf_string(scope, text)))
 
-#endif /* _PMTRACE_QMLPLUGIN_PROVIDER_H */
-
 #ifdef __cplusplus
 }
 #endif /*__cplusplus */
+
+#endif /* _PMTRACE_QMLPLUGIN_PROVIDER_H */
